Api/Src/Services/MaterialService.ts

```diff
@@ -0,0 +1,61 @@
+import * as express from "express";
+import * as mongoose from "mongoose";
+
+import MaterialRepository from "../Repositories/MaterialRepository";
+import MaterialSchema from "../Models/Material";
+import GradeSchema from "../Models/Grade";
+import SectionService from "./SectionService";
+import { TSection } from "../Models/Section";
+
+
+export default class MaterialService {
+
+    constructor(private repository: MaterialRepository, private sectionService: SectionService) {}
+
+    createMaterial = async ( req: express.Request ) => {
+        const id = new mongoose.Types.ObjectId(req.params.id);
+        const material = new MaterialSchema(req.body);
+        await material.validate();
+
+        let section: TSection = await this.sectionService.getSectionById(id);
```

Comment:
if (!section) throw Error("Section not found");

Reply:
added